### DISCUSSION TITLE
Umbraco 9 custom OWIN

### DISCUSSION THREAD

**Discussion by prosperva** [umbraco/Umbraco-CMS#11526] [Features and ideas]
I am trying to implement custom OWIN for Umbraco 9 and was wondering if this is possible. In our Umbraco 8, we are using LDAP to authenticate users.

**Comment by p-m-j** [comment]
OWIN doesn't exist as a concept in .net core, however there is a native middleware pipeline which you can configure in Startup.Configure.

https://docs.microsoft.com/en-us/aspnet/core/fundamentals/middleware/?view=aspnetcore-5.0
